pkg/restapi/kms/operation/operations.go

```diff
@@ -145,38 +140,89 @@ func (k kmsProvider) Crypto() crypto.Crypto {
 }
 
 func (o *Operation) createKeyHandler(rw http.ResponseWriter, req *http.Request) {
-	requestBody, err := ioutil.ReadAll(req.Body)
+	request := struct {
+		createKeyReq
+		Passphrase string `json:"passphrase"`
+	}{}
```

Comment:
I would create this as a struct in the models class, similar to createKeyReq

Reply:
the idea behind that is not to mix domain-related concept (that is a "create key" API, stable thing) with the infrastructure one (passphrase, that just happens to be our current mechanism to protect the key and will probably change in the future - so transient thing);

for now, I propose, to go with a half-way solution, like a dedicated struct for a "passphrase" param defined in the model - later I will try to get rid of that ugly mix of concerns (I was thinking on some sort of middleware - but I am still not sure about the exact implementation)